clients/scanoss/src/main/kotlin/model/BooleanSerializer.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2022 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.scanoss.model
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+/**
+ * SCANOSS returns booleans as "yes" or "no". Therefore, we need a special deserializer for it.
+ */
+class BooleanSerializer : KSerializer<Boolean> {
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor(
+        "SCANOSS_Boolean",
+        PrimitiveKind.BOOLEAN
+    )
+
+    override fun serialize(encoder: Encoder, value: Boolean) = encoder.encodeString(if (value) "yes" else "no")
+
+    override fun deserialize(decoder: Decoder): Boolean {
+        val string = decoder.decodeString()
+        return string == "yes"
```

Comment:
Nit: Maybe inline the `string` variable (and make this also a function expression).